rest/route/pod_agent.go

```diff
@@ -249,8 +252,62 @@ func (h *podAgentNextTask) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "marking pod as running"))
 	}
 
+	nextTaskResp := &apimodels.NextTaskResponse{}
 	h.setAgentFirstContactTime(p)
 
+	if p.Status != pod.StatusRunning {
+		nextTaskResp.ShouldExit = true
```

Comment:
We probably shouldn't ask the container agent to exit here. Can we instead enqueue the pod termination job?